locator/abstract_replication_strategy.cc

```diff
@@ -369,7 +369,7 @@ future<mutable_vnode_effective_replication_map_ptr> calculate_effective_replicat
             const auto token = all_tokens[i];
 
             auto current_endpoints = co_await rs->calculate_natural_endpoints(token, base_token_metadata);
-            auto target_endpoints = co_await rs->calculate_natural_endpoints(token, topology_changes->target_token_metadata);
+            auto target_endpoints = co_await rs->calculate_natural_endpoints(token, *topology_changes->target_token_metadata);
```

Comment:
Please add motivation for the patch (` token_metadata: topology_change_info: use token_metadata_ptr `) commit message.